SOCKET.io and ability to run any Python script given the filepath

samiam567 opened the pull request.

samiam567 committed:
Added SOCKET.io and ability to run a python script

Gjum commented:
I thought this integrated [Socket.io](http://socket.io/) / [Websockets](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) and was disappointed to see that it does not.

Other than that, your proposed code is highly specific to the setup on your machine, and only being able to send one message to the socket is very limited considering you are introducing a completely new API.

Instead of running just a Python script and logging its output (why?), a generic API to run system commands (including python scripts!) would be useful to have and maintain, and would be almost the same amount of code.

samiam567 commented:
I agree that the SOCKET io is underdeveloped. It was a primary implementation that I eventually abandoned. I guess you are correct that any functionality is not good functionality.

For the python script, it would be used (as I have on my machine) to send Minecraft events outside to your machine. The python script that is called can then do whatever to process that events. 
Upgrades to this implementation would be to:
a) add the ability to pass parameters to the python script
b) take the output (that is already collected) and return it in the Lua variable.

I used this functionality to send an event to an Arduino every time I took damage and figured I'd propose the additions to avoid wasting code.

samiam567 committed:
Update RunPythonScript.java

-added support for passing command line args to the python program

-function will now return a list of the output lines of the python program

samiam567 committed:
Update gradle.properties

MajsterTynek commented:
> a) add the ability to pass parameters to the python script
> b) take the output (that is already collected) and return it in the Lua variable.

inb4, why not use `io.popen` and use python executable installed locally?
